### PR TITLE
dash/kpi-render-event-bug

### PR DESCRIPTION
**Pull request by Vazonik**
Added render event emitters to the KPI Component.

### CODE REVIEW

File: ts/Dashboards/Components/KPIComponent.ts [hunk 1]
```diff
@@ -332,6 +332,7 @@ class KPIComponent extends Component {
 
 
     public render(): this {
+        this.emit({ type: 'beforeRender' });
```
Comment (bre1470) [inline ts/Dashboards/Components/KPIComponent.ts:335]: Not strictly related, but that is something, I already wondered in other components. We have `load` / `afterLoad`, `update` / `afterUpdate`, etc.; but here it is `beforeRender` / `afterRender`. We should be consistent and I think the best way to address this is actually to move the event as `render` into the base component class, so it gets emitted with the super call below like with the rest of events.
Any thoughts? @Vazonik , @sebastianbochan, @karolkolodziej , @pawellysy , @goransle

**Review by pawellysy — APPROVED** [review]
LGTM! Let's implement what @bre1470 suggested here as well tho.

File: ts/Dashboards/Components/KPIComponent.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/Dashboards/Components/KPIComponent.ts:335]: I agree :)

**Review by bre1470 — APPROVED** [review]
Nice!

**Review by karolkolodziej — APPROVED** [review]

**Review by goransle — APPROVED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**